agent_scheduler/task_helpers.py

```diff
@@ -142,32 +160,45 @@ def deserialize_img2img_image_args(args: dict):
 
 def serialize_controlnet_args(cnet_unit):
     args: dict = cnet_unit.__dict__
-    args["is_cnet"] = True
+    new_args = {}
+    new_args["is_cnet"] = True
     for k, v in args.items():
-        if k == "image" and v is not None:
-            args[k] = {
-                "image": serialize_image(v["image"]),
-                "mask": serialize_image(v["mask"])
-                if v.get("mask", None) is not None
-                else None,
-            }
-        if isinstance(v, Enum):
-            args[k] = v.value
+        if k == 'image':
+            if hasattr(v, "image") and v.image is not None:
+                new_args[k] = {
+                    "image": serialize_image(v.image),
+                    "mask": serialize_image(v.mask)
+                    if v.get("mask", None) is not None
+                    else None,
+                }
+            elif type(v) is dict and v.get('image', None) is not None:
+                new_args[k] = {
+                    "image": serialize_image(v["image"]),
+                    "mask": None if v.get("mask", None) is None else serialize_image(v["mask"]),
+                }
+            else:
+                new_args[k] = serialize_image(v)
+        elif isinstance(v, Enum):
+            new_args[k] = serialize_image(v.value)
+        else:
+            new_args[k] = serialize_image(v)
 
-    return args
+    return new_args
 
 
 def deserialize_controlnet_args(args: dict):
+    new_args = {}
     for k, v in args.items():
-        if k == "image" and v is not None:
-            args[k] = {
-                "image": deserialize_image(v["image"]),
-                "mask": deserialize_image(v["mask"])
-                if v.get("mask", None) is not None
-                else None,
-            }
+        if k == "image" and isinstance(v, dict) and v.get("image", None) is not None:
+            new_args["image"] = deserialize_image(v["image"])
```

Comment:
The `image` should be a `Dict` in format:
```
{ "image": ..., "mask": "" }
```
ControlNet also supports direct `image` as top level property, but `mask` will be ignored.